pyteal/ast/expr.py

```diff
@@ -1,20 +1,35 @@
 from abc import ABC, abstractmethod
-from typing import Tuple, List, TYPE_CHECKING
+from typing import TYPE_CHECKING, List, Tuple
 
-from pyteal.types import TealType
 from pyteal.ir import TealBlock, TealSimpleBlock
+from pyteal.stack_frame import Frames
+from pyteal.types import TealType
 
 if TYPE_CHECKING:
     from pyteal.compiler import CompileOptions
 
 
+def frames_adder(init):
+    def wrapper(*args, **kwargs):
+        args[0].frames = Frames()
+
+        return init(*args, **kwargs)
+
+    return wrapper
+
+
 class Expr(ABC):
     """Abstract base class for PyTeal expressions."""
 
+    def __init_subclass__(cls):
+        cls.__init__ = frames_adder(cls.__init__)
```

Comment:
With this trick, even subclasses that don't call `__super__().__init__(...)` can get frames added.